flor/utils.py

```diff
@@ -56,3 +58,17 @@ def get_partitions(iterator, num_gpu):
         partitions.append(iterator[i * work_per_gpu: (i + 1) * work_per_gpu])
         i += 1
     return partitions
+
+
+def deepcopy_cpu(x):
+    if hasattr(x, 'cpu'):
+        return x.cpu()
+
+    if isinstance(x, dict):
+        x_copy = {}
+        for k, v in x.items():
+            x_copy[k] = deepcopy_cpu(v)
+        assert id(x_copy) != id(x)
+        return x_copy
+
```

Comment:
There's a bug here. You also need to do a deep copy when x is not a dict. You should fall back to copy.deepcopy().